server/queue/fifo.go

```diff
@@ -254,28 +255,36 @@ func (q *fifo) KickAgentWorkers(agentID int64) {
 }
 
 // helper function that loops through the queue and attempts to
-// match the item to a single subscriber.
+// match the item to a single subscriber until context got cancel.
 func (q *fifo) process() {
-	q.Lock()
-	defer q.Unlock()
+	for {
+		q.Lock()
+		if q.paused {
```

Comment:
Imho this will use a lot of CPU cycles of paused as it will loop and lock/unlock cpnstantly

Reply:
ah dammed :/

I had the time wait at the start first but moved it back for the tests